### DISCUSSION TITLE
Closing Drawer when item is clicked in navigation menu

### DISCUSSION THREAD

**Discussion by sylcha** [quasarframework/quasar#12652] [General - Components / Directives / etc] [answered]
Hi there,

I look for a way to close a drawer whenever an element in the navigation menu list is clicked (knowing that `@click` is already used for a link...). I tried to add `v-close-popup` to each item but it does not help...

Does anyone have an idea ?

Sylvain

**Comment by MilosPaunovic** [comment]
[https://codepen.io/milospaunovic/pen/RwjvwGN?editors=1010](https://codepen.io/milospaunovic/pen/RwjvwGN?editors=1010)

**Reply by sylcha** [reply]
Yes, I use this trick for the menu button. But as I said, the `@click` directive is already used by a fonction. In my case, it updates something using a store action. In addition the drawer is declared in `MainLayout` and I need to close it from my component...

-----
**My Vue component**
```
<template>
  <q-item
    clickable
    tag="a"
    target="_parent"
    :to="link"
    exact
    @click="updateAppName({ newName: title })"
  >
    <q-item-section v-if="icon" avatar>
      <q-icon :name="icon" />
    </q-item-section>

    <q-item-section>
      <q-item-label>{{ title }}</q-item-label>
      <q-item-label caption>
        {{ caption }}
      </q-item-label>
    </q-item-section>
  </q-item>
</template>

<script>
import { defineComponent } from "vue";
import { mapActions } from "vuex";

export default defineComponent({
  name: "EssentialLink",
  props: {
    title: {
      type: String,
      required: true,
    },

    caption: {
      type: String,
      default: "",
    },

    link: {
      type: String,
      default: "#",
    },

    icon: {
      type: String,
      default: "",
    },
  },
  methods: {
    ...mapActions("names", ["updateAppName"]),
  },
});
</script>
```

-------

**Part of my `MainLayout` layout**

```
    <q-drawer
      v-model="leftDrawerOpen"
      behavior="desktop"
      :show-if-above="false"
      bordered
    >
      <q-list>
        <q-item-label header> Applications </q-item-label>

        <EssentialLink
          v-for="link in essentialLinks"
          :key="link.title"
          v-bind="link"
        />
      </q-list>
    </q-drawer>
```

**Comment by MilosPaunovic** [comment, marked as answer]
If you're using store, either `Vuex` or `Pinia`, the best solution would be to create a property in state which will be the model for your drawer and a mutation which will toggle that property. That way, you can import it and call from any component.

**Reply by sylcha** [reply]
It's working! Thanks for your solution! I did not know that a `@click` could take more than one argument.

Following your advice, I added a `openMenu` property to my state. I get it with a `isMenuOpen` getter. The drawer use this `isMenuOpen` as a model. Then, I added a `TOGGLE_MENU` mutation that I command with a `toggleMenu` action. That way, I am able to do something like the following that works like a charm!

```
@click="
      updateAppName({ newName: title });
      toggleMenu();
    "
```
